src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -110,12 +110,14 @@ const predicate_map = {
 
 const condition_fit_functions = {
     's3:ExistingObjectTag': _is_object_tag_fit,
-    's3:x-amz-server-side-encryption': _is_server_side_encryption_fit
+    's3:x-amz-server-side-encryption': _is_server_side_encryption_fit,
+    's3:VersionId': _is_object_version_fit
 };
 
 const supported_actions = {
```

Comment:
Could you add the link that you worked with as a comment before `supported_actions`?
https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-policy-keys

Reply:
Added the above link which mention all the other keys also